hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSink.java

```diff
@@ -95,6 +95,10 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
       pipeline = Pipelines.hoodieStreamWrite(conf, parallelism, hoodieRecordDataStream);
       // compaction
       if (OptionsResolver.needsAsyncCompaction(conf)) {
+        // batch mode write must use syncCompaction.
+        if (context.isBounded()) {
+          conf.setBoolean(FlinkOptions.COMPACTION_ASYNC_ENABLED, false);
```

Comment:
In streaming exec mode, a bounded source would also triggers checkpoint, should we disable the async compaction for them ?

Reply:
My idea is that when the source is bounded, we should not do compaction in checkpoint, because compaction will be done once in `endinput`. Am I right?

Reply:
Not exactly, because the bounded source can also be long running.